File: repos_estudiantes_TARQUI_segunda_entrega/RUBIN_DE_CELIS_VALDEZ_ANDRES/HERENCIA/EJERCICIO-3/Persona.py
from datetime import datetime

class Persona:
    def __init__(self, ci="", nombre="", apellido="", celular="", fecha_Nac=""):
        self.ci = ci
        self.nombre = nombre
        self.apellido = apellido
        self.celular = celular
        self.fecha_Nac = fecha_Nac

    def calcularEdad(self):
        nacimiento = datetime.strptime(self.fecha_Nac, "%d/%m/%Y")
        hoy = datetime.now()
        edad = hoy.year - nacimiento.year - ((hoy.month, hoy.day) < (nacimiento.month, nacimiento.day))
        return edad

    def mostrar(self):
        print(f"CI: {self.ci}, Nombre: {self.nombre}, Apellido: {self.apellido}, Celular: {self.celular}, Fecha Nac: {self.fecha_Nac}, Edad: {self.calcularEdad()}")

File: repos_estudiantes_TARQUI_segunda_entrega/Quispe_Huanca_Carlos_Daniel/HERENCIA/EJERCICIO-3/Persona/src/ejercicio3/Persona.java
package ejercicio3;

import java.time.LocalDate;
import java.time.Period;
import java.time.format.DateTimeFormatter;

public class Persona {
    protected String ci;
    protected String nombre;
    protected String apellido;
    protected String celular;
    protected String fechaNac;
    protected String sexo;
    
    public Persona() {
        this("", "", "", "", "2000-01-01", "");
    }
    
    public Persona(String ci, String nombre, String apellido, String celular, String fechaNac, String sexo) {
        this.ci = ci;
        this.nombre = nombre;
        this.apellido = apellido;
        this.celular = celular;
        this.fechaNac = fechaNac;
        this.sexo = sexo;
    }
    
    public int calcularEdad() {
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
        LocalDate fechaNacimiento = LocalDate.parse(fechaNac, formatter);
        LocalDate ahora = LocalDate.now();
        return Period.between(fechaNacimiento, ahora).getYears();
    }
    
    public void mostrar() {
        System.out.println("CI: " + ci);
        System.out.println("Nombre: " + nombre);
        System.out.println("Apellido: " + apellido);
        System.out.println("Celular: " + celular);
        System.out.println("Fecha Nacimiento: " + fechaNac);
        System.out.println("Sexo: " + sexo);
        System.out.println("Edad: " + calcularEdad() + " años");
    }
    
    public String getCi() { return ci; }
    public String getNombre() { return nombre; }
    public String getApellido() { return apellido; }
    public String getCelular() { return celular; }
    public String getFechaNac() { return fechaNac; }
    public String getSexo() { return sexo; }
    
    public void setCi(String ci) { this.ci = ci; }
    public void setNombre(String nombre) { this.nombre = nombre; }
    public void setApellido(String apellido) { this.apellido = apellido; }
    public void setCelular(String celular) { this.celular = celular; }
    public void setFechaNac(String fechaNac) { this.fechaNac = fechaNac; }
    public void setSexo(String sexo) { this.sexo = sexo; }
}